iot/api/CloudIotMqttExampleTest/CloudIotMqttTest.cs

```diff
@@ -0,0 +1,262 @@
+﻿// Copyright(c) 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+using Google.Cloud.Iam.V1;
+using Google.Cloud.PubSub.V1;
+using Grpc.Core;
+using Xunit;
+using System;
+using Google.Api;
+using System.IO;
+using Xunit.Abstractions;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace GoogleCloudSamples
+{
+    // <summary>
+    /// Runs the sample app's methods and tests the outputs
+    // </summary>
+    public class CommonTests : IClassFixture<IotTestFixture>
+    {
+        private readonly RetryRobot _retryRobot = new RetryRobot()
+        {
+            RetryWhenExceptions = new[] { typeof(Xunit.Sdk.XunitException) }
+        };
+        private readonly IotTestFixture _fixture;
+        private readonly ITestOutputHelper _output;
+
+        public CommonTests(IotTestFixture fixture, ITestOutputHelper helper)
+        {
+            _fixture = fixture;
+            //for displaying unit tests output in the console.
+            _output = helper;
+        }
+
+        ConsoleOutput Run(params string[] args) => _fixture.Run(args);
+
+        private void Eventually(Action action) => _retryRobot.Eventually(action);
+
+        //[START iot_mqtt_tests]
```

Comment:
Will this code be rendered on a page somewhere?

Reply:
No, not at this moment.

Reply:
Ok.  Then it doesn't need a doc tag.  Doc tags are exclusively for code that will be rendered to a page.